core/src/test/scala/org/apache/spark/deploy/rest/SubmitRestProtocolSuite.scala

```diff
@@ -86,6 +86,8 @@ class SubmitRestProtocolSuite extends SparkFunSuite {
     message.clientSparkVersion = "1.2.3"
     message.appResource = "honey-walnut-cherry.jar"
     message.mainClass = "org.apache.spark.examples.SparkPie"
+    message.appArgs = Array("hdfs://tmp/auth")
+    message.environmentVariables = Map("SPARK_HOME" -> "/test")
```

Comment:
There should be a check at the end of this test to make sure these fields show up in the `newMessage` object. (around L.125)

Reply:
and probably don't want to set SPARK_HOME either? :)

Reply:
@felixcheung okay done :)

Reply:
@susanxhuynh I've checked what you said but those variables are overwritten below, so I can't actually check it, right?

Reply:
@Gschiavon You're right, I didn't see those variables were already set below. But, I do have a question now about whether these fields are optional or not. In the original test, (L.93), there's a comment about "optional fields", and the app args and env vars are tested there (L.105-106). Since this test was added with Standalone mode in mind, I wonder if these fields are considered optional there? There's no documentation as you said so we can't know for sure. I just want to make sure the extra validation doesn't break any existing applications (in Standalone cluster mode).

Reply:
@susanxhuynh Yes we have to make sure we don't break the Standalone mode.
I've reviewed StandaloneSubmitRequestServlet class and i think we might be facing the same problem here, (L.126-131) You can find checks for 'appResource' and 'mainClass' but not for 'appArgs'(L.140) or 'environmentVariables'(L.143) when they could be null as they are initialised as null.

I think is the same case, let me know what you think.

Reply:
OK, I was looking at RestSubmissionClient (used by Mesos and Standalone) - the client does always set 'appArgs' and 'envrionmentVariables': https://github.com/apache/spark/blob/master/core/src/main/scala/org/apache/spark/deploy/rest/RestSubmissionClient.scala#L427 So, it's only a bug when the user does not use the RestSubmissionClient (by using 'curl' directly to the server, for example). So, that addresses my concern about Standalone mode. As to whether to add a similar check in the Standalone class, I don't have a strong opinion about it (fix it here or fix in another PR).

Reply:
Perfect then @susanxhuynh . Fix it maybe in another PR?

Reply:
Yeah, that's fine with me.

Reply:
Cool, I can do it when I have some time.

Let me know if I have to something else here!

Thanks @susanxhuynh

Reply:
LGTM